File: modules/crypto-core/src/main/java/com/shterneregen/securelan/crypto/workflow/HybridFileCryptoWorkflow.java
package com.shterneregen.securelan.crypto.workflow;

import com.shterneregen.securelan.crypto.model.HybridEncryptedData;

import java.security.PrivateKey;
import java.security.PublicKey;

public interface HybridFileCryptoWorkflow {
    HybridEncryptedData encrypt(byte[] fileBytes, PublicKey publicKey);

    byte[] decrypt(HybridEncryptedData encryptedData, PrivateKey privateKey);
}

File: modules/crypto-core/src/main/kotlin/com/shterneregen/securelan/crypto/workflow/HybridFileCryptoWorkflow.kt
package com.shterneregen.securelan.crypto.workflow

import com.shterneregen.securelan.crypto.model.HybridEncryptedData
import java.security.PrivateKey
import java.security.PublicKey

interface HybridFileCryptoWorkflow {
    fun encrypt(fileBytes: ByteArray, publicKey: PublicKey): HybridEncryptedData

    fun decrypt(encryptedData: HybridEncryptedData, privateKey: PrivateKey): ByteArray
}
